### ISSUE TITLE
Swagger route shows enum names but response schema does not

### ISSUE THREAD

**Issue by Brutiquzz** [unchase/Unchase.Swashbuckle.AspNetCore.Extensions#11]
I have installed 
Unchase.Swashbuckle.AspNetCore.Extensions v2.3.8
Swashbuckle.AspNetCore v5.5.1

When using swagger I am trying to make it show me the enum names both when filling parameters but also when reading the response schema. Just like your sample picture for a SamplePerson title illustrates it. 

However I am only getting the results you are showing for parameters but not for the response schema. 
![image](https://user-images.githubusercontent.com/33770011/87161666-239c1980-c2c5-11ea-91e4-cc568ad91c70.png)

I have pretty much copy pasted the code from your example.
![image](https://user-images.githubusercontent.com/33770011/87162019-a329e880-c2c5-11ea-9ab5-932aec7242d2.png)

Can you tell me what I am missing or doing wrong?

**Comment by unchase** [comment]
Hi, @Brutiquzz 
Can you send me your generated specification json-file (or part of it) to spiritkola@hotmail.com?
I will check what the problem is.

**Comment by unchase** [comment]
@Brutiquzz 
Can you try again with the latest version ([v 2.3.10](https://www.nuget.org/packages/Unchase.Swashbuckle.AspNetCore.Extensions/2.3.10))?

**Comment by hahn-kev** [comment]
I'm also seeing this issue :( with version 2.3.10. Target framework core app 3.1, same swashbuckle version as OP.

**Comment by unchase** [comment]
> I'm also seeing this issue :( with version 2.3.10. Target framework core app 3.1, same swashbuckle version as OP.

Hi, @hahn-kev

Can you show me your json-specification?

**Comment by hahn-kev** [comment]
I've edited it down a bit, to make it simpler for you and because I'm not sure how public it's allowed to be.
```json
{
  "openapi": "3.0.1",
  "info": {
    "title": "API",
    "version": "v1"
  },
  "paths": {
    "/v1/payments": {
      "post": {
        "tags": [
          "Payments"
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/AuthorizePaymentBody"
              }
            }
          },
          "required": true
        },
        "responses": {
          "200": {
            "description": "Success",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/WebshopAuthorizePaymentResponse"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "AuthorizePaymentBody": {
        "required": [
          "Amount"
        ],
        "type": "object",
        "properties": {
          "amount": {
            "type": "number",
            "description": "Order amount",
            "format": "double"
          }
        },
        "additionalProperties": false
      },
      "PaymentErrorCode": {
        "enum": [
          0,
          1,
          2
        ],
        "type": "integer",
        "format": "int32",
        "x-enumNames": [
          "noError",
          "typeMissMatch",
          "unexpected"
        ]
      },
      "WebshopAuthorizePaymentResponse": {
        "type": "object",
        "properties": {
          "errorCode": {
            "$ref": "#/components/schemas/PaymentErrorCode"
          }
        },
        "additionalProperties": false,
        "example": {
          "ErrorCode": 0
        }
      }
    }
  }
}
```

**Comment by unchase** [comment]
Hi, @hahn-kev 

I can't reproduce the issue with your json-specification:

![success](https://user-images.githubusercontent.com/29679226/91842464-c84f2d80-ec5c-11ea-83c0-ce492aaa1b72.png)

Can you show your controller code?

**Comment by hahn-kev** [comment]
I can't as this is for a closed source project. What's serving the Swagger UI you showed in your screenshot? Looking at the `versions` object in the console my swagger UI version is 3.25 maybe you're testing with a newer version?

**Comment by unchase** [comment]
I used 3.26 version.

**Comment by unchase** [comment]
> I can't as this is for a closed source project. What's serving the Swagger UI you showed in your screenshot? Looking at the `versions` object in the console my swagger UI version is 3.25 maybe you're testing with a newer version?

You can share only the part of controller code. For example, only the method signature.

**Comment by hahn-kev** [comment]
Sure, it's something like this
`Task<ActionResult<WebshopAuthorizePaymentResponse>> AuthorizePayment(
            [FromBody] AuthorizePaymentBody checkout)`

**Comment by unchase** [comment]
@hahn-kev @Brutiquzz 
I created a [repository with a test project](https://github.com/unchase/Unchase.Swashbuckle.AspNetCore.Extensions.Issue11) for the issue, in which the description for the enum is displayed correctly. Hope it helps you.

**Comment by ghost** [comment]
I have a similar issue, please see attached swagger.txt
[swagger.txt](https://github.com/unchase/Unchase.Swashbuckle.AspNetCore.Extensions/files/5246049/swagger.txt)

![image](https://user-images.githubusercontent.com/49510009/93608573-0f625000-f9cb-11ea-8909-4c7f5d478649.png)


````csharp

services.AddSwaggerGen(c =>
            {
                c.AddEnumsWithValuesFixFilters();
            });
````

and
````csharp
 app.UseSwaggerUI(c =>
            {
                c.SwaggerEndpoint("/swagger/v1/swagger.json", "Internal DropShip API");
            });
````

````xml
<PackageReference Include="Swashbuckle.AspNetCore" Version="5.6.1" />
<PackageReference Include="Unchase.Swashbuckle.AspNetCore.Extensions" Version="2.3.11" />
````

**Comment by unchase** [comment]
Hi, @IainCoSource 

Should use:

```csharp
services.AddSwaggerGen(c =>
{
    c.AddEnumsWithValuesFixFilters();

    var xmlFilePath = Path.Combine(AppContext.BaseDirectory, "<your_project_name>.xml");
    options.IncludeXmlComments(xmlFilePath);
});
```

with setting:

![xml](https://user-images.githubusercontent.com/29679226/93750116-a9144200-fc03-11ea-885f-ca2d995e6798.png)

**Comment by hahn-kev** [comment]
I thought we only needed to include that if we want to include XML comments, is that required for the Enum values?

**Comment by ghost** [comment]
@unchase I implemented the following

````csharp
  var mvcXmlFilePath = Path.Combine(AppContext.BaseDirectory, "Pims.Mvc.xml");
  c.IncludeXmlComments(mvcXmlFilePath);

  var enumXmlFilePath = Path.Combine(AppContext.BaseDirectory, "Updater4.Data.xml");
  c.IncludeXmlComments(enumXmlFilePath);
````

The first one has added some comments, the second does nothing for the enums.

![image](https://user-images.githubusercontent.com/49510009/93798161-869f1a80-fc3d-11ea-8261-1740898469e8.png)

example of the compiler generated xml.

````xml
 <member name="F:Updater4.Data.Pims.DropShipOrderStatusType.SupplierError">
            <summary>
            Supplier Error.
            </summary>
            <remarks>IsError: true.</remarks>
        </member>
````

**Comment by unchase** [comment]
@Brutiquzz @hahn-kev @IainCoSource
Fixed in [v 2.3.12](https://www.nuget.org/packages/Unchase.Swashbuckle.AspNetCore.Extensions/2.3.12)

You can use:

```csharp
services.AddSwaggerGen(c =>
{
    c.AddEnumsWithValuesFixFilters();
});
```

**Comment by ghost** [comment]
@unchase  :) New version is awesome

![image](https://user-images.githubusercontent.com/49510009/93802705-18118b00-fc44-11ea-9a9a-52949c663a28.png)
